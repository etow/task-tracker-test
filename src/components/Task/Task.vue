<template>
  <div class="task">
    <div>{{ task.name }}</div>
    <div v-if="task.estimate" class="task__estimate">
      {{ task.estimate }}
    </div>
  </div>
</template>

<script>
export default {
  props: {
    task: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
  .task {
    border-radius: 15px;
    background-color: #fff;
    padding: 20px;
    cursor: pointer;
    box-shadow: rgba(0, 0, 0, 0.02) 0px 0px 0px 1px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px, rgba(0, 0, 0, 0.05) 0px 2px 8px 0px;
    display: flex;
    &__estimate {
      background-color: #eaeaea;
      border-radius: 5px;
      padding: 2px 5px;
      margin-left: auto;
      font-size: 12px;
    }
  }
</style>
