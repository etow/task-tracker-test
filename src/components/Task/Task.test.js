import { mount } from '@vue/test-utils'
import Task from './Task.vue'

describe('Project.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(Task, {
      propsData: {
        task: { name: 'TASK NAME' }
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
