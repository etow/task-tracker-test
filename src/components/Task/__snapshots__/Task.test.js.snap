// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project.vue renders correctly 1`] = `
<div class="task">
  <div>TASK NAME</div>
  <!--v-if-->
</div>
`;
